extensions/amp-iframe/1.0/component.js

```diff
@@ -30,6 +35,42 @@ export function Iframe({
   const isIntersectingRef = useRef(null);
   const containerRef = useRef(null);
 
+  const viewabilityCb = (entries) => {
+    const iframe = iframeRef.current;
+    if (!iframe) {
+      return;
+    }
+    const win = toWin(iframe.ownerDocument.defaultView);
+    postMessage(
+      iframe,
+      MessageType.INTERSECTION,
+      cloneEntryForCrossOrigin(entries[0]),
+      win.location.origin
+    );
+  };
+
+  useEffect(() => {
+    const iframe = iframeRef.current;
+    if (!iframe) {
+      return;
+    }
+    const win = toWin(iframe.ownerDocument.defaultView);
+    let observer;
+    win.addEventListener('message', (event) => {
+      if (event.data?.type !== MessageType.SEND_INTERSECTIONS) {
```

Comment:
Need to verify the sender is the iframe's window.

Reply:
Done.